csharp/extractor/Semmle.Extraction.CSharp.Standalone/BuildAnalysis.cs

```diff
@@ -434,12 +486,16 @@ private void AnalyseSolutions(IEnumerable<string> solutions)
             });
         }
 
-        public void Dispose()
-        {
-            packageDirectory?.Dispose();
-        }
+        public void Dispose() => packageDirectory?.Dispose();
 
-        [GeneratedRegex("<PackageReference .*Include=\"(.*?)\".*/>", RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.Singleline)]
+        [GeneratedRegex("<PackageReference\\s+Include=\"(.*?)\".*/?>", RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.Singleline)]
         private static partial Regex PackageReference();
+
+        [GeneratedRegex("<FrameworkReference\\s+Include=\"(.*?)\".*/?>", RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.Singleline)]
+        private static partial Regex FrameworkReference();
+
+        [GeneratedRegex("<Project\\s+Sdk=\"(.*?)\".*/?>", RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.Singleline)]
```

Comment:
I think this regex needs some slight tweaking. There are some optional XML attributes that could come between the opening tag and `Sdk=`. For example the below is also valid:
```
<Project ToolsVersion="15.0" Sdk="Microsoft.NET.Sdk.Web">
```

Reply:
It might be a rare case, but it's possible to define the SDK in at least one other way. We should probably cover the below case too: 
```
<Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk.Web" />
```

Reply:
Right! I will change the regexp back to what it approximately except that we want to match the whitespace using `\s`. I at least found one case where this was needed for the matching to be succesfull.